### PR TITLE
#2561 Comments refinement

### PR DESCRIPTION
**Pull request by ccanos**
### Describe the background of your pull request

What does your pull request do? Does it solve a bug (which one?), add a feature?

### Additional context

Add any other context

### Governance 

- [ ] Documentation is added
- [ ] Test cases are added or updated

By submitting this pull request I confirm that:
- I've read the contributing document https://github.com/alkem-io/.github/blob/master/CONTRIBUTING.md.
- I've read and understand the Code of Conduct https://github.com/alkem-io/.github/blob/master/CODE_OF_CONDUCT.md.
- I understand that any contributions or suggestions I made may make it into the actual code. I've read the License 
     https://github.com/alkem-io/Coordination/blob/master/LICENSE.
- I understand that submitting the pull request requires agreeing to and signing the contributor license agreement.

### CODE REVIEW

File: src/domain/aspect/views/AspectDashboardView.tsx [hunk 1]
```diff
@@ -171,7 +168,6 @@ const AspectDashboardView: FC<AspectDashboardViewProps> = props => {
                   placeholder={t('pages.aspect.dashboard.comment.placeholder')}
                   onPostComment={onPostComment}
                   maxLength={MID_TEXT_LENGTH}
-                  userAvatarUri={user?.user?.profile?.avatar?.uri}
```
Comment (hero101) [inline src/domain/aspect/views/AspectDashboardView.tsx:174]: why was this required?

File: src/domain/communication/useAuthorsDetails.ts [hunk 2]
```diff
@@ -3,26 +3,89 @@ import { Author } from '../../models/discussion/author';
...
   const { data: authorData, loading } = useAuthorDetailsQuery({
     variables: { ids: uniqIds },
     skip: uniqIds.length === 0,
   });
+  const { hubId } = useHub();
+  const { challengeId } = useChallenge();
+  const { opportunityId } = useOpportunity();
+
+  const resourceId = opportunityId || challengeId || hubId || '';
+  const reourceType: ResourceTypeName = opportunityId ? 'opportunity' : challengeId ? 'challenge' : 'hub';
```
Comment (hero101) [inline src/domain/communication/useAuthorsDetails.ts:68]: typo in var name

File: src/models/constants/translation.constants.ts [hunk 3]
```diff
@@ -1,5 +1,8 @@
 export const OWNER_TRANSLATION_KEY = 'common.owner';
 export const ADMIN_TRANSLATION_KEY = 'common.admin';
 export const MEMBER_TRANSLATION_KEY = 'common.member';
+export const LEAD_TRANSLATION_KEY = 'common.lead';
+export const HOST_TRANSLATION_KEY = 'common.host';
 
 export type RoleNameKey = typeof OWNER_TRANSLATION_KEY | typeof ADMIN_TRANSLATION_KEY | typeof MEMBER_TRANSLATION_KEY;
+export type AuthorRoleNameKey = typeof LEAD_TRANSLATION_KEY | typeof HOST_TRANSLATION_KEY;
```
Comment (hero101) [inline src/models/constants/translation.constants.ts:8]: this does not give enough context to where it's used. `[Comment/Message]Author` gives more context imo.

**Review by hero101 — CHANGES_REQUESTED** [review]
My main issues are things missed on the last PR about comments.
In general I think that the "comments" code must stop living under Discussions.
For example:
- the comment component is not a "Discussions comment"
- the author avatar component should not be specially built for discussions (or live under discussions)

File: src/domain/aspect/views/AspectDashboardView.tsx (hunk 1 above)
Comment (ccanos) [reply src/domain/aspect/views/AspectDashboardView.tsx:174]: Before the restyling the `PostComment` component was just a Markdown without the user avatar in it. After Aleks's changes it has the avatar on the left of the input field.
This is some code that I let Aleks merge to refactor it later because Neil told us to close Aleks'es task and let me continue on this other one. 
You can see his changes here: https://github.com/alkem-io/client-web/pull/2549/files
He added the userContext to the AspectDashboardView  only to pass the userAvatar to the PostComment form field. 
I didn't like that very much so I have moved that code to the PostComment.

After my last changes PostComment is renamed to PostMessageToCommentsForm

File: src/domain/communication/useAuthorsDetails.ts (hunk 2 above)
Comment (ccanos) [reply src/domain/communication/useAuthorsDetails.ts]: fixed, thank you

File: src/models/constants/translation.constants.ts (hunk 3 above)
Comment (ccanos) [reply src/models/constants/translation.constants.ts]: `MessageAuthorRoleNameKey` I like it more too

**Comment by ccanos** [comment]
@hero101 I have refactored and moved all Comments stuff to `domain/shared/components/Comments` 
I like that more too. Changed a bit some names to make proper distinction between a Comments object, and a Message
